### PR TITLE
EZP-30344: Improved handling Language Limitation

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-30344
| Requires | ezsystems/ezpublish-kernel#2585
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    |no
| Tests pass?   | yes
| Doc needed?   |no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

This PR improves handling of `Language` Limitation based on logic implemented in ezsystems/ezpublish-kernel#2585

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/lib/Form/Type/ChoiceList/Loader/ContentEditTranslationChoiceLoader.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     * @param \EzSystems\EzPlatformAdminUi\Permission\LookupLimitationsTransformer $lookupLimitationsTransformer
+     * @param string[] $languageCodes
+     */
+    public function __construct(
+        LanguageService $languageService,
+        PermissionResolver $permissionResolver,
+        ?ContentInfo $contentInfo,
+        LookupLimitationsTransformer $lookupLimitationsTransformer,
+        $languageCodes
```
Comment (adamwojs) [inline src/lib/Form/Type/ChoiceList/Loader/ContentEditTranslationChoiceLoader.php]:
```suggestion
        array $languageCodes
```

File: src/lib/Form/Type/ChoiceList/Loader/ContentEditTranslationChoiceLoader.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        if (!empty($this->languageCodes)) {
+            $languages = array_filter(
+                $languages,
+                function (Language $language) {
+                    return \in_array($language->languageCode, $this->languageCodes, true);
+                }
+            );
+        }
+        $languagesCodes = array_column($languages, 'languageCode');
```
Comment (adamwojs) [inline src/lib/Form/Type/ChoiceList/Loader/ContentEditTranslationChoiceLoader.php]:
```suggestion

        $languagesCodes = array_column($languages, 'languageCode');
```

File: src/lib/Permission/LookupLimitationsTransformer.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+     *
+     * @return array
+     */
+    public function getFlattenedLimitationsValues(LookupLimitationResult $lookupLimitations): array
+    {
+        $limitationsValues = [];
+
+        /** @var \eZ\Publish\API\Repository\Values\User\LookupPolicyLimitations $lookupPolicyLimitation */
+        foreach ($lookupLimitations->lookupPolicyLimitations as $lookupPolicyLimitation) {
+            /** @var Limitation $limitation */
```
Comment (adamwojs) [inline src/lib/Permission/LookupLimitationsTransformer.php]:
```suggestion
            /** @var \eZ\Publish\API\Repository\Values\User\Limitation $limitation */
```

File: src/bundle/Controller/ContentController.php [hunk 4]
```diff
@@ -456,4 +473,47 @@ public function updateVisibilityAction(Request $request): Response
...
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param string|null $languageCode
+     *
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function checkEditPermissionAction(Content $content, ?string $languageCode): JsonResponse
```
Comment (adamwojs) [inline src/bundle/Controller/ContentController.php]: TODO: Make sure response is not cached by varnish 😉

File: src/lib/Menu/ContentEditRightSidebarBuilder.php [hunk 5]
```diff
@@ -97,8 +98,9 @@ public function createStructure(array $options): ItemInterface
         /** @var Location $parentLocation */
         $parentLocation = $options['parent_location'];
 
-        $canPublish = $this->permissionResolver->canUser('content', 'publish', $content);
-        $canEdit = $this->permissionResolver->canUser('content', 'edit', $content);
+        $version = (new Version())->translateToAnyLanguageOf([$language->languageCode]);
```
Comment (alongosz) [inline src/lib/Menu/ContentEditRightSidebarBuilder.php]: Let's not forget that this is a target (I get what happens, but I doubt anyone beside us will ;) ):
```suggestion
        $target = (new Target\Version())->translateToAnyLanguageOf([$language->languageCode]);
```

File: src/bundle/Resources/public/js/scripts/admin.version.edit.conflict.js [hunk 6]
```diff
@@ -1,22 +1,45 @@
...
 
-    [...doc.querySelectorAll('.ez-btn--content-draft-edit')].forEach(link => link.addEventListener('click', editVersion, false));
-})(window, document, window.jQuery);
+        fetch(checkEditPermissionLink, { mode: 'same-origin', credentials: 'same-origin' })
+            .then(eZ.helpers.request.getJsonFromResponse)
+            .then(handleCanEditCheck)
+            .then(handleVersionDraftConflict)
+            .catch(showErrorNotification);
+    };
+    doc.querySelectorAll('.ez-btn--content-draft-edit').forEach((button) => button.addEventListener('click', editVersion, false));
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.version.edit.conflict.js]: empty line after `const`

File: src/bundle/Resources/public/js/scripts/admin.version.edit.conflict.js [hunk 7]
```diff
@@ -1,22 +1,45 @@
...
+})(window, document, window.jQuery, window.eZ);
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/admin.version.edit.conflict.js]: empty line at the end of file

**Pull request merged**